feat(Header): addressing accessibility concerns and replacing subheader with description

mnajdova opened the pull request:
# Header

Addressing accessibility concerns and replacing the subheader prop with description. Previous discussions: https://github.com/stardust-ui/react-old/pull/123

### TODO

- [ ] Conformance test
- [x] Minimal doc site example
- [ ] Stardust base theme
- [x] Teams Light theme
- [ ] Teams Dark theme
- [ ] Teams Contrast theme
- [ ] Confirm RTL usage
- [x] [W3 accessibility](https://www.w3.org/standards/webdesign/accessibility) check
- [ ] [Stardust accessibility](https://github.com/stardust-ui/accessibility) check
- [ ] Update glossary props table
- [x] Update the CHANGELOG.md

levithomason commented on src/components/Icon/Icon.js:
```diff
@@ -0,0 +1,114 @@
+    //   useValueAndKey(rotated, 'rotated'),
+    //   'icon',
+    //   className,
+    // )
```
Safe to delete all the cruft here.  We'll reimplement what we need as we need it.

levithomason commented on src/components/Icon/Icon.js:
```diff
@@ -0,0 +1,114 @@
+    // size: PropTypes.oneOf(_.without(SUI.SIZES, 'medium')),
+
+    /** Icon can have an aria label. */
+    'aria-label': PropTypes.string,
```
Let's strip propTypes down to only what this PR will add.  Preferably, the bare minimum needed to render the default example.  Probably the `name` prop?

We'll add props and examples as we need them.

kuzhelov commented:
taking this PR to split it on two different ones

Commit:
-added subheaderAs prop
-restructured the DOM elements of the Header component

Commit:
-reverted the changes
-added default value for the as property of the subheader
-added texAlign example

Commit:
-reverted default as value for the subheader to be div

kuzhelov-ms committed:
introduce desired html markup

kuzhelov-ms committed:
Merge branch 'master' into header/addresing-accessibility-concerns

kuzhelov-ms committed:
remove enclosing div from html

kuzhelov-ms committed:
Merge branch 'master' into header/addresing-accessibility-concerns

kuzhelov-ms committed:
register changes in CHANGELOG

Commit:
-replaced Header subheader with description

Commit:
-improved changelog description

Commit:
Merge branch 'master' into header/addresing-accessibility-concerns-origin

# Conflicts:
#	CHANGELOG.md
#	src/components/Header/Header.tsx

Commit:
-changed header example description

mnajdova committed:
Merge branch 'master' into header/addresing-accessibility-concerns

levithomason commented:
See the new repo Icon.  We've trimmed it down and updated it there for our initial needs.

https://github.com/stardust-ui/react
